apps/test/unit/templates/teacherDashboard/teacherNavigation/TeacherNavigationBarTest.tsx

```diff
@@ -0,0 +1,63 @@
+import {render, screen} from '@testing-library/react';
+import React from 'react';
+import {Provider} from 'react-redux';
+import {Store, AnyAction} from 'redux';
+
+import {getStore, registerReducers} from '@cdo/apps/redux';
+import teacherSections, {
+  serverSectionFromSection,
+  setSections,
+} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
+import TeacherNavigationBar from '@cdo/apps/templates/teacherNavigation/TeacherNavigationBar';
+import i18n from '@cdo/locale';
+
+describe('TeacherNavigationBar', () => {
+  const sections = [
+    {
+      id: 11,
+      name: 'Period 1',
+      hidden: false,
+    },
+    {
+      id: 12,
+      name: 'Period 2',
+      hidden: false,
+    },
+    {
+      id: 13,
+      name: 'Period 3',
+      hidden: true,
+    },
+  ];
+  const serverSections = sections.map(serverSectionFromSection);
+
+  let store: Store<unknown, AnyAction>;
+
+  beforeEach(() => {
+    store = getStore();
+    registerReducers({
+      teacherSections,
+    });
+    store.dispatch(setSections(serverSections));
+  });
+
+  test('renders correctly with visible sections', () => {
+    render(
+      <Provider store={store}>
+        <TeacherNavigationBar />
+      </Provider>
+    );
+
+    const classLabel = screen.getByText(i18n.classSections());
+    expect(classLabel).toBeDefined();
+
+    const dropdown = screen.getByRole('combobox');
+    expect(dropdown).toBeDefined();
```

Comment:
We shouldn't need this line because `getByText` already throws an error if it isn't found. Same with classLabel above and options below
```suggestion
```

Reply:
I have always had trouble just leaving `screen.getByRole` as the sole test that something is in the document.  I understand, technically, it is just as useful as using an assertion, but assertions feel so much more readable to me and the intention feels clear.  This is probably in part a hold over from chai. 

Have you seen any standards as a team that would prefer one over another?  I noticed a lot of Denys' components to follow the pattern of getting component and then asserting they are there.  Since he has been the most consistent contributor of TypeScript components with tests I was following his lead.  @levadadenys do you have any insights about conversations around this through your work?

Again, happy to agree to disagree (re: readability and clarity), just trying to get a sense of the bigger trend/sentiment in the org.